### DISCUSSION TITLE
Useful snippets / cookbook

### DISCUSSION THREAD

**Discussion by Kypsis** [GregoryConrad/rearch-dart#112] [General]
One thing I would love to see more from package makers (especially if they are unidiomatic to the ecosystem) is useful snippets / cookbook how they use the package themselves or solutions to common use-cases in documentation. Eg navigation and route-guards, data fetching and mutations (eg disabled _SignIn_ button after pressing it and the mutation is in progress), persistence. For Rearch things like (lazy)`stateGetterSetter`, `reducer`, `mutation` (this one took me more time to grok than expected due to wrong assumptions on my part), other useful tricks and patterns. 

Rearch in a way is quite unidiomatic to Flutters very heavily class based approach vs the more functional style. Quite natural to pick up if you have used React before but maybe not so much for devs without such a background. There is an appreciable opportunity cost to using Rearch. So snippets / cookbook might make the sell easier.

I used the words snippets / cookbook intentionally as full blown example can be way too much work for the package maintainers as well. And some useful patterns or snippets might be easy to pull from previous professional implementations when otherwise the work to either untangle proprietary part or creating the full context around it might be prohibitive.

**Comment by GregoryConrad** [comment]
> Rearch in a way is quite unidiomatic to Flutters very heavily class based approach vs the more functional style. 
> Quite natural to pick up if you have used React before but maybe not so much for devs without such a background.

It definitely is different for those only used to OOP, but for those familiar with FP it should be easier to pick up. I've actually never used React before 🥲

> So snippets / cookbook might make the sell easier.

Definitely; I have some ideas on the back burner but it'll take awhile before I can get to it all.

I am currently tracking this via https://github.com/GregoryConrad/rearch-docs/issues/12, but it may be awhile before I can get around to writing it. Maybe I can make a flow-chart that helps pick what side effect to use when making a capsule/how to combine side effects

**Comment by GregoryConrad** [comment]
I realized that snippets may not fit well into the page I had in mind, so I just made a new page for them

https://rearch.gsconrad.com/snippets
